### PR TITLE
feat: add auto updater functionality

### PR DESCRIPTION
**Pull request by goga-m**
## Summary
- [x] Monitor and push notification for new release
- [x] Implement `download-update`
- [x] Integrate download progress in `WalletUpdate` modal
- [x] Implement `download-complete` action
- [x] Implement `quit-and-install` action
- [x] Fix differential download issue. Show downloading status with spinner when download progress is not emitted

## Checklist

- [ ] Documentation _(if necessary)_
- [x] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

**Comment by faustbrian** [comment]
Task linked: [CU-2f3ch5 Add auto updater functionality](https://app.clickup.com/t/2f3ch5)

File: package.json [hunk 1]
```diff
@@ -235,6 +236,8 @@
 		"electron-devtools-installer": "^3.1.1",
 		"electron-notarize": "^1.0.0",
 		"electron-root-path": "^1.0.16",
+		"electron-updater": "^4.3.5",
+		"electron-log": "^4.2.4",
```
Comment (luciorubeens) [inline package.json:240]: These must be non-dev dependencies

File: src/electron/updater.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+const { autoUpdater } = require("electron-updater");
+const { version } = require("../../package.json");
+const logger = require("electron-log");
+
+const setupConfig = () => {
+	autoUpdater.logger = logger;
+	autoUpdater.logger.transports.file.level = "info";
+	autoUpdater.autoDownload = false;
+	autoUpdater.currentVersion = version;
+	autoUpdater.updateConfigPath = "app-update.yml";
```
Comment (luciorubeens) [inline src/electron/updater.js:10]: This is causing error not found in production mode. E2E can be detected with `process.env.ELECTRON_IS_E2E`.

**Review by luciorubeens — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (goga-m) [reply package.json]: Also moved `async-retry` as non-dev dep, as it was causing not-found error in production.

File: src/electron/updater.js (hunk 2 above)
Comment (goga-m) [reply src/electron/updater.js]: Included `ELECTRON_IS_E2E` check to add `app-update.yml`. Seems to be working ok now in dev and e2e. Thanks!

**Review by luciorubeens — APPROVED** [review]

File: src/electron/updater.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		process.env.APPIMAGE = `dist/target/ark-desktop-wallet-linux-x86_64-${testVersion || version}.AppImage`;
+	}
+};
+
+const setupConfig = ({ isDev, testVersion, isE2e }) => {
+	autoUpdater.logger = logger;
+	autoUpdater.logger.transports.file.level = "info";
+
+	autoUpdater.autoDownload = false;
+	autoUpdater.currentVersion = version;
```
Comment (goga-m) [inline src/electron/updater.js:18]:
```suggestion
	autoUpdater.currentVersion = testVersion || version;
```

**Pull request merged**